spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,149 @@
+  end
+
+  def initial_page_stub
+    from = Time.zone.today - 30.days
+    to = Time.zone.today
+    content_data_api_has_metric(base_path: 'base/path',
+                                from: from,
+                                to: to,
+                                metrics: metrics)
+    content_data_api_has_timeseries(base_path: 'base/path',
+                                    from: from,
+                                    to: to,
+                                    metrics: metrics,
+                                    payload: {
+       unique_pageviews: [
+         { "date" => (from - 1.day).to_s, "value" => 0 },
+         { "date" => (from - 2.days).to_s, "value" => 9 },
+         { "date" => (to + 1.day).to_s, "value" => 9 }
+       ],
+       pageviews: [
+         { "date" => (from - 1.day).to_s, "value" => 8 },
+         { "date" => (from - 2.days).to_s, "value" => 8 },
+         { "date" => (to + 1.day).to_s, "value" => 8 }
+       ],
+       number_of_internal_searches: [
+         { "date" => (from - 1.day).to_s, "value" => 8 },
+         { "date" => (from - 2.days).to_s, "value" => 8 },
+         { "date" => (to + 1.day).to_s, "value" => 8 }
+       ],
+       satisfaction_score: [
+         { "date" => (from - 1.day).to_s, "value" => 100 },
+         { "date" => (from - 2.days).to_s, "value" => 90 },
+         { "date" => (to + 1.day).to_s, "value" => 80 }
+       ]
+     })
+  end
+
+  def stub_response_for_date_range(from, to)
+    content_data_api_has_metric(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: metrics)
+    content_data_api_has_timeseries(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: metrics,
+      payload: {
+        unique_pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 1 },
+          { "date" => (from - 2.days).to_s, "value" => 2 },
+          { "date" => (to + 1.day).to_s, "value" => 30 }
+        ],
+        pageviews: [
+          { "date" => (from - 1.day).to_s, "value" => 10 },
+          { "date" => (from - 2.days).to_s, "value" => 20 },
+          { "date" => (to + 1.day).to_s, "value" => 30 }
+        ],
+        number_of_internal_searches: [
+          { "date" => (from - 1.day).to_s, "value" => 8 },
+          { "date" => (from - 2.days).to_s, "value" => 8 },
+          { "date" => (to + 1.day).to_s, "value" => 8 }
+        ],
+        satisfaction_score: [
+          { "date" => (from - 1.day).to_s, "value" => 100 },
+          { "date" => (from - 2.days).to_s, "value" => 90 },
+          { "date" => (to + 1.day).to_s, "value" => 80 }
+        ]
+      })
+  end
+
+  def expect_metrics_for_each_date_to_be_correct(from, to)
+    month_and_date_string_for_date1 = (from - 1.day).to_s.last(5)
+    month_and_date_string_for_date2 = (from - 2.days).to_s.last(5)
+    month_and_date_string_for_date3 = (to + 1.day).to_s.last(5)
+    unique_pageviews_rows = find("#unique_pageviews_table").all('tr')
```

Comment:
Could you please reuse @peteglondon helper for 
getting all the values from a table?